src/profile/profile.service.ts

```diff
@@ -63,6 +64,38 @@ export class ProfileService {
     }
   }
 
+  /**
+   * It updates a profile by id
+   * @param {string} profileId - string - The id of the profile to be deleted
+   * @param {UpdateProfileDto} updateProfileDto - UpdateProfileDto - the data to be updated
+   * @returns The delete method returns a DeleteResult object.
+   */
+
+  async updateProfile(
+    profileId: string,
+    updateProfileDto: UpdateProfileDto,
+  ): Promise<Profile> {
+    try{
+      const profile = await this.profileRepo.findOne({
+        where: { id: profileId },
+      });
+      if (profile) {
+        await this.profileRepo.update(profileId, updateProfileDto);
+  
+        const undatedProfile = await this.profileRepo.findOne({ where: {id:profileId}});
```

Comment:
Please what is this check for ???

Reply:
@danifara3 please answer

Reply:
It supplies the updated if it has been actually updated the FE can used the details there to confirm